### DISCUSSION TITLE
Disable auto focus for option

### DISCUSSION THREAD

**Discussion by khawerrind** [JedWatson/react-select#5447] [Ideas]
First of all thank you for the great library. 

I wanted to ask if there is any option to not select any option by default? I want the option to be selected only when user use arrow keys or click on an option using mouse.

**Comment by tung-dang** [comment]
In this code line https://github.com/JedWatson/react-select/blob/master/src/Select.js#L458, look like 
`focusOption` is always the first or last menu option. 
I am also looking for a way to disable this behaviour - "auto select an option when menu open".

**Comment by software-project** [comment]
Anyone found a solution for this?

**Comment by yurii-lubynets** [comment]
Hello, @JedWatson 
Thank you for the flexible and useful tool. Could you provide some hardcoded solution for this issue? And what about the `autoFocus` prop?
Thanks,
Yurii

**Comment by alisman** [comment]
While it would be great for this PR to be merged, I believe I managed to accomplish this with the following hackery:

onInputChange={(value: string) =>{ this.selectRef.select.getNextFocusedOption = ()=>null; }}

There is no longer any default selection, but arrow keying still works.

I then also bind to onKeyDown to intercept submission from input and fire submission callback with text directly from input.

**Reply by jasondainter** [reply]
This worked great for me.

**Reply by EamonNerbonne** [reply]
`getNextFocusedOption` doesn't seem available anymore in v5; I now use this: https://github.com/JedWatson/react-select/pull/4080#issuecomment-2367902345

Do either of you have a more up-to-date fix?

**Comment by manvydasu** [comment]
There is an approved PR for this - https://github.com/JedWatson/react-select/pull/3010

**Comment by bladey** [comment]
This will be resolved when #4080 is merged.

**Comment by augusto-proano** [comment]
@khawerrind, quick question, are you planning to merge this pr at some point?

**Comment by dayoolacodes** [comment]
is there a fix for this yet? this [pr](https://github.com/JedWatson/react-select/pull/4080) has not been merged yet

**Comment by zulqarnain-empg** [comment]
Hi! any update on this? This is a must-have functionality, kindly merge the above PR.

Does any one have any solution to this?

**Reply by tasawernawaz** [reply]
@zulqarnain-empg meanwhile you can use this https://github.com/JedWatson/react-select/discussions/5447#discussioncomment-4014386

**Reply by zulqarnain-empg** [reply]
> @zulqarnain-empg meanwhile you can use this [#5447 (comment)](https://github.com/JedWatson/react-select/discussions/5447#discussioncomment-4014386)

using a package called 'react-select-async-paginate' and can't find getNextFocusedOption in its ref

**Reply by tasawernawaz** [reply]
I passed two props to simple <Select> like `ref={props.selectRef}`, `onInputChange={props.onInputChange}`, then in `onInputChange`, i did the following

```
       if (this.selectRef.current) {
                // by default it focuses first element
                (this.selectRef.current.select as any).getNextFocusedOption = () => null;
            }
```

**Reply by zulqarnain-empg** [reply]
unfortunately couldn't find getNextFocusedOption anywhere in react-select-async-paginate's ref

**Reply by EamonNerbonne** [reply]
I didn't try to disable auto-focus entirely, but simply to reset the focus when the list of options changes.  That seems to work with this workaround: https://github.com/JedWatson/react-select/pull/4080#issuecomment-2367902345

**Comment by Appolinars** [comment]
Hi everyone. Any updates about this? After all these years I see that PRs connected to preventing auto-focusing on first option are still opened or closed. None of them merged. So there is no built-in prop\solution to prevent this auto-focusing? 
Some people provide this solution:
onInputChange={(value: string) =>{ this.selectRef.select.getNextFocusedOption = ()=>null; }}
But it looks like workaround and kind of dirty solution. Maybe there are better approaches?

**Comment by PratikAwaik** [comment]
Can someone merge relevant PR related to this? Need this feature.
@JedWatson
